### PR TITLE
Support stabilised syntax for OpenQASM 3 `switch`

### PR DESCRIPTION
**Pull request by jakelishman**
### Summary

The OpenQASM 3 project made some changes to the preliminary syntax of the `switch` statement, merging the final version in openqasm/openqasm#463.  This commit adds support for the new syntax, making it available by default while still supporting the experimental flag to enable the preliminary syntax.  Support for the old flag is necessary to allow Qiskit to continue to interoperate with other tooling that still assumes the preliminary format (we know that at least the IBM QSS compiler is in this position right now), and in general should hopefully not be too onerous to continue to support.

### Details and comments

The new syntax won't be supported by the QSS compiler yet, and there's no need to rush to that - it's easy for Qiskit to continue to support both, and having an easy flag switch over means that any deployments of Qiskit alongside other OpenQASM 3 parsers can easily keep them in sync without having to fall behind on Qiskit bugfixes.

### CODE REVIEW

File: qiskit/qasm3/ast.py [hunk 1]
```diff
@@ -596,11 +596,30 @@ class DefaultCase(Expression):
     """An object representing the `default` special label in switch statements."""
 
 
-class SwitchStatement(Statement):
-    """AST node for the proposed 'switch-case' extension to OpenQASM 3."""
+class SwitchStatementV1(Statement):
+    """AST node for the proposed 'switch-case' extension to OpenQASM 3.
+
+    The stabilised form of the syntax instead uses :class:`.SwitchStatement`."""
```
Comment (mtreinish) [inline qiskit/qasm3/ast.py]:
```suggestion
    The stabilized form of the syntax instead uses :class:`.SwitchStatement`."""
```

File: qiskit/qasm3/ast.py [hunk 2]
```diff
@@ -596,11 +596,30 @@ class DefaultCase(Expression):
     """An object representing the `default` special label in switch statements."""
 
 
-class SwitchStatement(Statement):
-    """AST node for the proposed 'switch-case' extension to OpenQASM 3."""
+class SwitchStatementV1(Statement):
```
Comment (mtreinish) [inline qiskit/qasm3/ast.py]: I'd be inclined to call this something a bit more descriptive like `SwitchStatementPreview` or `ExperimentalSwitchStatement`. When I see V1 here I feel like this means there is another stable revision in the spec.

**Review by mtreinish — COMMENTED** [review]
This LGTM, just one :gb: nit inline. The other thing is the name of the statement class for the experimental version. But both are fairly minor.

File: qiskit/qasm3/ast.py (hunk 2 above)
Comment (jakelishman) [reply qiskit/qasm3/ast.py]: Yeah, I took the V1 from the name in the `ExperimentalFlags` enum, but `SwitchStatementPreview` or something is better.  Deon in 089165901d.

File: releasenotes/notes/qasm3-stable-switch-61a10036d1587778.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  - |
+    The OpenQASM 3 exporter (see :func:`~.qasm3.dump` and :func:`~.qasm3.dumps` functions in
+    :mod:`qiskit.qasm3`) now supports the stabilised syntax of the ``switch`` statement in OpenQASM 3
+    by default.  The pre-certification syntax of the ``switch`` statement is still available by
+    using the :attr:`.ExperimentalFeatures.SWITCH_CASE_V1` flag in the ``experimental`` argument of
+    the exporter.  There is no feature flag required for the stabilised syntax, but if you are
+    interfacing with other tooling that is not yet updated, you may need to pass the experimental
+    flag.
+
+    The syntax of the stabilised form is slightly different with regards to terminating ``break``
```
Comment (mtreinish) [inline releasenotes/notes/qasm3-stable-switch-61a10036d1587778.yaml:12]:
```suggestion
    :mod:`qiskit.qasm3`) now supports the stabilized syntax of the ``switch`` statement in OpenQASM 3
    by default.  The pre-certification syntax of the ``switch`` statement is still available by
    using the :attr:`.ExperimentalFeatures.SWITCH_CASE_V1` flag in the ``experimental`` argument of
    the exporter.  There is no feature flag required for the stabilized syntax, but if you are
    interfacing with other tooling that is not yet updated, you may need to pass the experimental
    flag.

    The syntax of the stabilized form is slightly different with regards to terminating ``break``
```

**Review by mtreinish — APPROVED** [review]
I caught some :gb: "stabilised" in the release note, but we can just fix that in the release note round-up for the final release. Thanks for the quick update.

File: qiskit/qasm3/ast.py [hunk 4]
```diff
@@ -596,11 +596,31 @@ class DefaultCase(Expression):
     """An object representing the `default` special label in switch statements."""
 
 
-class SwitchStatement(Statement):
-    """AST node for the proposed 'switch-case' extension to OpenQASM 3."""
+class SwitchStatementPreview(Statement):
+    """AST node for the proposed 'switch-case' extension to OpenQASM 3, before the syntax was
```
Comment (jlapeyre) [inline qiskit/qasm3/ast.py:600]: I'm trying to think of a ~better~ more useful name than `....Preview`. I think contemporary software devs find the word "Legacy" distasteful. But it's probably widely understood by casual programmers. Maybe `SwitchStatementOldStyle`. In OQ3, "old style" refers to OQ2 legacy syntax. In this case it's not from a previous version. But `OldStyle` tells me that I probably don't want this, even if I am not familiar with its history.

Or even `SwitchStatementObsolete`. This really tells me I should not be using it.

**Pull request merged**